app/views/propertyLinking/declaration.scala.html

```diff
@@ -16,20 +16,30 @@
 
 @import binders.propertylinks._
 @import controllers.propertyLinking.DeclarationVM
+@import controllers.propertyLinking.ClaimPropertyOwnershipController
 @import config.ApplicationConfig
 @import uk.gov.hmrc.govukfrontend.views.html.components._
 @import uk.gov.hmrc.govukfrontend.views.html.components.implicits._
-@this(govukErrorSummary : GovukErrorSummary, govukWarningText: GovukWarningText, govukCheckboxes: GovukCheckboxes, govukButton : GovukButton, mainLayout: views.html.mainLayout, formWithCSRF: FormWithCSRF)
-@(model: DeclarationVM, isRatesBillEvidence: Boolean, clientDetails: Option[ClientDetails] = None)(implicit request: Request[_], messages: Messages, config: ApplicationConfig)
+@import java.time.format.DateTimeFormatter
+
+@this(govukErrorSummary : GovukErrorSummary, govukWarningText: GovukWarningText, govukCheckboxes: GovukCheckboxes, govukButton : GovukButton, mainLayout: views.html.mainLayout, formWithCSRF: FormWithCSRF, govukSummaryList: GovukSummaryList)
+
+
+@(model: DeclarationVM, isRatesBillEvidence: Boolean, clientDetails: Option[ClientDetails] = None, propertyRelationship: Option[PropertyRelationship], propertyOwnership: Option[PropertyOwnership], uploadEvidenceData: UploadEvidenceData)(implicit request: Request[_], messages: Messages, config: ApplicationConfig)
 
 @backLink = @{if (isRatesBillEvidence) {
                     controllers.propertyLinking.routes.UploadController.show(EvidenceChoices.RATES_BILL).url
                 } else {
                     controllers.propertyLinking.routes.UploadController.show(EvidenceChoices.OTHER).url
                 }
 }
+
 @clientId = @{clientDetails.map(_.organisationId)}
 
+@strippedFileName = @{uploadEvidenceData.fileInfo.get.name.replaceAll("^([^-]+)-", "")}
+
+@evidenceType = @{uploadEvidenceData.fileInfo.get.evidenceType.get.name}
```

Comment:
`.get.` followed by another `.get.` this code will explode as soon as there's no `fileInfo`, or the `fileInfo` has no `evidenceType`.